### PR TITLE
upcoming: [M3-7874] - Linode Create Refactor - More Marketplace Progress

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Makes more progress on the Marketplace tab of Linode Create v2 🔧 

## Changes  🔄
- Introduces `oneClickAppsv2.ts` so that there is one source of truth (previously both `oneClickAppsv.ts` and `stackScriptUtils.ts` would have to be referenced)
- Makes new `AppDetailsDrawer` that accepts a `stackscriptId` instead of a label. I'm not sure why we were using labels...

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Notice how there are duplicate apps and broken icons | Notice how expected apps are rendered |
| Screenshot 2024-05-20 at 12 58 48 PM | Screenshot 2024-05-20 at 12 58 39 PM |

### Prerequisites
- Enable Linode Create v2 feature flag 🇺🇸

### Verification steps
- Verify you can now view app details
- Verify that the apps rendered match the apps rendered in production

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by hkhalil-akamai — APPROVED** [review]
Looking clean and verified parity with prod.

**Review by hkhalil-akamai — CHANGES_REQUESTED** [review]
I forgot to switch on the flag 🤦🏽‍♂️

It appears that app icons don't display when using the preview link:

Screenshot 2024-05-20 at 5 45 31 PM

**Comment by bnussman-akamai** [comment]
Good catch @hkhalil-akamai

Fixed in 1c7d215

File: packages/manager/src/features/OneClickApps/oneClickApps.ts
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/OneClickApps/oneClickApps.ts:1]: I have no idea 😞 

I think the new format (`oneClickAppsv2`) will make things considerably more simple. If you think of anything cleaner, let me know! I'm open to anything! Now is the time to fix all of this stuff

[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/OneClickApps/oneClickApps.ts:1]: For additional context, here's what I'm hoping for.

**The old setup sources data from**: StackScripts API, flags.oneClickApps, flags.oneClickAppsDocsOverride, `stackScriptUtils.ts`, and `oneClickApps.ts`

**The new setup will source data from**: StackScripts API, `oneClickAppsv2.ts`

We could keep a feature flag around for emergencies if we need to append to `oneClickAppsv2.ts`, but maybe we can go without it at first. cc @hana-linode

**Review by jaalah-akamai — COMMENTED** [review]
Do we have a plan to filter the apps by `New apps`, `Popular apps`, `All apps`?

**Comment by jaalah-akamai** [comment]
Are we sunsetting the logo and information icon when an app is selected? The new design omits the both.
 
Screenshot 2024-05-21 at 10 15 14 AM

**Comment by bnussman-akamai** [comment]
@jaalah-akamai All of that will come! I'm trying to do things in small PRs

**Review by hkhalil-akamai — APPROVED** [review]
Confirmed app icons are displaying. Nice work, Marketplace is coming along really well!

**Review by jaalah-akamai — APPROVED** [review]
✅ Confirmed new experience fixes issues listed
✅ Confirmed other implementation details will soon follow

**Pull request merged**